src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs

```diff
@@ -102,14 +110,67 @@ internal BlameCollector(IBlameReaderWriter blameReaderWriter, IProcessDumpUtilit
             if (this.configurationElement != null)
             {
                 var collectDumpNode = this.configurationElement[Constants.DumpModeKey];
-                this.processDumpEnabled = collectDumpNode != null;
-                if (this.processDumpEnabled)
+                this.collectProcessDumpCollectionBasedOnTrigger = collectDumpNode != null;
+                if (this.collectProcessDumpCollectionBasedOnTrigger)
                 {
```

Comment:
all of this code can go into the BlameConfigurationsProcessor (which can be a separate class in itself)  and returns a DiagnosticsConfiguration object which will have details about the dump collection mode, results directory etc..

Reply:
Thought of this too, had actual done this partially in the other abandoned PR i had. But again don't have a lot of time to spend on this.